### PR TITLE
Add support for parsing to debug tags

### PR DESCRIPTION
**Pull request by hwang381**
This pull request add support for parsing console output lines as "debug tags". An image preview is attached below.
capture
Test has also been fixed as mentioned in PR #11

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/logparser/LogParserWorkflowTest.java [hunk 1]
```diff
@@ -40,7 +40,7 @@ public void logParserPublisherWorkflowStep() throws Exception {
         job.setDefinition(new CpsFlowDefinition(""
                         + "node {\n"
                         + "  def mvnHome = tool '" + mavenInstallation.getName() + "'\n"
-                        + "  sh \"${mvnHome}/bin/mvn clean install\"\n"
+                        + "  sh \"export M2_HOME=${mvnHome};${mvnHome}/bin/mvn clean install\"\n"
```
Comment (recena) [inline src/test/java/org/jenkinsci/plugins/logparser/LogParserWorkflowTest.java]: Unrelated change.

**Comment by recena** [comment]
Please, add a specific test for this new feature.

**Comment by hwang381** [comment]
I have reverted back the test.
As for testing this new feature, since I am just adding parsing support for another tag, I put assertion for number of debug tags in the existing test (which happens to be 0). How shall I proceed to make a specific test for it? Thanks.

**Comment by recena** [comment]
@hwang381 For example, providing a output with debugs messages and checking the result.

**Comment by hwang381** [comment]
Using Mockito we will be able to creates new instance of AbstractBuild that has two [debug] tags in this build result.

**Comment by recena** [comment]
@hwang381 An alternative [here](https://github.com/jenkinsci/log-parser-plugin/blob/master/src/test/java/org/jenkinsci/plugins/logparser/LogParserWorkflowTest.java#L36). I used the project with its output, in this case a Maven project.

**Comment by hwang381** [comment]
Can I just have the same test as logParserPublisherWorkflowStep() but only asserting the amount of debug tags as 0? Or is it better if I have another project zip that can generate more than zero debug tags? It seems hard to generate debug tags by project since they are internal to maven according to my investigation.

**Comment by hwang381** [comment]
Hi @recena, here are some explanations on the recent changes with the test.
1. The Maven project extraction & execution logic has been moved into the init method, and at the end it will set the global variable "result".
2. This "result" variable is then used by both test methods
3. getClass() on line 32 is replaced by LogParserWorkflowTest.class since getClass() cannot be invoked in static context.
